### DISCUSSION TITLE
Is it possible to use Jinja2 Templates?

### DISCUSSION THREAD

**Discussion by chrisbodon** [vitalik/django-ninja#522] [Q&A]
I installed Jinja2 as its supposed to be installed in a Django project. But I cant render the templates in any way. Its there a exact method to achieve this?

**Comment by vitalik** [comment]
@chrisbodon 

sure simply return http response with rendered jinja2 html

```Python
from django.http import HttpResponse

@api.get('/some')
def some(request):
    html = jinja2_redner(...
    return HttpResponse(html)
```

**Reply by chrisbodon** [reply]
Exactly what I needed. Thank you very much for answering so quickly.

**Reply by chrisbodon** [reply]
It worked, but I need to access the templates from http://127.0.0.1:8000/, and not from http://127.0.0.1:8000/api because that its what i want only for api purposes. Do I need to create another api = NinjaAPI() and add it to the urlpatterns? For example:

api = NinjaAPI()
app = Something()

urlpatterns = [
	path("admin/", admin.site.urls),
	path("api/", api.urls),
	path("/", app.urls)
]

Again, thank you very much for your time.

**Reply by vitalik** [reply]
I do not think you need a django-ninja in that case - you should simply use default django rendering but just set Jinja2 as templates backend if you need to use jinja2

https://stackoverflow.com/a/30715508/208525

**Reply by chrisbodon** [reply]
I see. But, in the case of having 2 django apps, let's say: api and api2, if I incorporate the urls of both apps (where api = NinjaApi() and api2 = NinjaApi()), and it should be ok, right? I mean, thre should be no problem on separating or having multiple instances of NinjaApi across different apps, right? That was also part of my question (that maybe I was not so clear about, sorry for that).
